packages/svelte/src/compiler/phases/3-transform/utils.js

```diff
@@ -222,7 +225,7 @@ export function infer_namespace(namespace, parent, nodes, path) {
 					only_svg = false;
 					break;
 				}
-			} else if (node.type !== 'Text' || node.data.trim() !== '') {
+			} else if (node.type !== 'ConstTag' && (node.type !== 'Text' || node.data.trim() !== '')) {
```

Comment:
Thank you for tackling this, but this is not quite right. We need to keep the check above as-is, and instead need to check that if we discover if/await/each blocks while traversing the nodes here, that the top level children of those are all svg elements (basically this loop needs to become recursive). Also, could you add a test case for this?

Reply:
Yeah, I realized that is the correct approach while trying to fix it, that's why I kept it a draft. I will add a test case shortly and maybe implement this.